### PR TITLE
fix(ep): remove rga non-schematized search cause

### PR DESCRIPTION
**Pull request by louis-bompart**
https://coveord.atlassian.net/browse/KIT-3404

### CODE REVIEW

**Review by alexprudhomme — APPROVED** [review]
What happens whenever a 'next' event is not specified ?

**Comment by louis-bompart** [comment]
> What happens whenever a 'next' event is not specified ?

Good question: no actionCause will be sent.
This is the conclusion we reached with the DataPlatform team

**Review by mmitiche — APPROVED** [review]
👍

File: packages/headless/src/features/generated-answer/generated-answer-analytics-actions.ts [hunk 1]
```diff
@@ -408,10 +408,6 @@ export const retryGeneratedAnswer = (): SearchAction => ({
   actionCause: SearchPageEvents.retryGeneratedAnswer,
 });
 
-export const rephraseGeneratedAnswer = (): SearchAction => ({
-  actionCause: SearchPageEvents.rephraseGeneratedAnswer,
-});
-
```
Comment (SimonMilord) [inline packages/headless/src/features/generated-answer/generated-answer-analytics-actions.ts:414]: do we not want to remove the retry as well here?

**Comment by louis-bompart** [comment]
Following the slack discussion, @mmitiche @SimonMilord , I rerequest your review.
I think this PR is ready to be scrubbed maybe 🤔

**Comment by mmitiche** [comment]
@louis-bompart  Following the same slack discussion: https://coveo.slack.com/archives/C02ET4TQDD4/p1721921559026459?thread_ts=1721918413.953499&cid=C02ET4TQDD4, ServiceData/DataPlatform gang agreed to add the `retryGeneratedAnswer` and `rephraseGeneratedAnswer` to the schemas. 

In that case wouldn't better to avoid removing these two causes now just to re-add them in like one week when they will be added to the accepted schemas?

**Comment by louis-bompart** [comment]
> @louis-bompart Following the same slack discussion: https://coveo.slack.com/archives/C02ET4TQDD4/p1721921559026459?thread_ts=1721918413.953499&cid=C02ET4TQDD4, ServiceData/DataPlatform gang agreed to add the `retryGeneratedAnswer` and `rephraseGeneratedAnswer` to the schemas.
> 
> In that case wouldn't better to avoid removing these two causes now just to re-add them in like one week when they will be added to the accepted schemas?

yes.
Pending update, will reloop on the PR after the changes in the schema.

**Comment by louis-bompart** [comment]
Trying to dust off this PR @mmitiche, so essentially I should revert the changes that touches RGA, right?

**Comment by mmitiche** [comment]
> Trying to dust off this PR @mmitiche, so essentially I should revert the changes that touches RGA, right?

Exactly yeah @louis-bompart , the RGA related search causes have been added to the schemas 👍

**Comment by louis-bompart** [comment]
Third read, pretty certain everything has been standardized.
Closing.

**Pull request closed without merging**